CMakeLists.txt

```diff
@@ -0,0 +1,127 @@
+set(CHPL_HOST_COMPILER CACHE STRING "Description")
+set(CHPL_HOST_CC CACHE STRING "Description")
+set(CHPL_HOST_CXX CACHE STRING "Description")
+set(CHPL_HOST_BUNDLED_COMPILE_ARGS CACHE STRING "Description")
+set(CHPL_HOST_SYSTEM_COMPILE_ARGS CACHE STRING "Description")
+set(CHPL_HOST_BUNDLED_LINK_ARGS CACHE STRING "Description")
+set(CHPL_HOST_SYSTEM_LINK_ARGS CACHE STRING "Description")
+set(CHPL_HOST_ARCH CACHE STRING "Description")
+set(CHPL_HOST_CPU CACHE STRING "Description")
+set(CHPL_TARGET_COMPILER_PRGENV CACHE STRING "Description")
+set(CHPL_TARGET_BUNDLED_COMPILE_ARGS CACHE STRING "Description")
+set(CHPL_TARGET_SYSTEM_COMPILE_ARGS CACHE STRING "Description")
+set(CHPL_TARGET_BUNDLED_LINK_ARGS CACHE STRING "Description")
+set(CHPL_TARGET_SYSTEM_LINK_ARGS CACHE STRING "Description")
+set(CHPL_RUNTIME_CPU CACHE STRING "Description")
+set(CHPL_TARGET_CPU_FLAG CACHE STRING "Description")
+set(CHPL_TARGET_BACKEND_CPU CACHE STRING "Description")
+set(CHPL_HOST_MEM CACHE STRING "Description")
+set(CHPL_HOST_JEMALLOC CACHE STRING "Description")
+set(CHPL_TARGET_MEM CACHE STRING "Description")
+set(CHPL_TARGET_JEMALLOC CACHE STRING "Description")
+set(CHPL_MAKE CACHE STRING "Description")
+set(CHPL_GMP CACHE STRING "Description")
+set(CHPL_GMP_IS_OVERRIDDEN CACHE STRING "Description")
+set(CHPL_RE2_IS_OVERRIDDEN CACHE STRING "Description")
+set(CHPL_LLVM CACHE STRING "Description")
+set(CHPL_LLVM_SUPPORT CACHE STRING "Description")
+set(CHPL_LLVM_CONFIG CACHE STRING "Description")
+set(CHPL_LLVM_VERSION CACHE STRING "Description")
+set(CHPL_LLVM_CLANG_C CACHE STRING "Description")
+set(CHPL_LLVM_CLANG_CXX CACHE STRING "Description")
+set(CHPL_SANITIZE CACHE STRING "Description")
+set(CHPL_RUNTIME_SUBDIR CACHE STRING "Description")
+set(CHPL_LAUNCHER_SUBDIR CACHE STRING "Description")
+set(CHPL_COMPILER_SUBDIR CACHE STRING "Description")
+set(CHPL_HOST_BIN_SUBDIR CACHE STRING "Description")
+set(CHPL_TARGET_BIN_SUBDIR CACHE STRING "Description")
+set(CHPL_SYS_MODULES_SUBDIR CACHE STRING "Description")
+set(CHPL_LLVM_UNIQ_CFG_PATH CACHE STRING "Description")
+set(CHPL_GASNET_UNIQ_CFG_PATH CACHE STRING "Description")
+set(CHPL_GMP_UNIQ_CFG_PATH CACHE STRING "Description")
+set(CHPL_HWLOC_UNIQ_CFG_PATH CACHE STRING "Description")
+set(CHPL_HOST_JEMALLOC_UNIQ_CFG_PATH CACHE STRING "Description")
+set(CHPL_TARGET_JEMALLOC_UNIQ_CFG_PATH CACHE STRING "Description")
+set(CHPL_LIBFABRIC_UNIQ_CFG_PATH CACHE STRING "Description")
+set(CHPL_LIBUNWIND_UNIQ_CFG_PATH CACHE STRING "Description")
+set(CHPL_QTHREAD_UNIQ_CFG_PATH CACHE STRING "Description")
+set(CHPL_RE2_UNIQ_CFG_PATH CACHE STRING "Description")
+set(CHPL_PE_CHPL_PKGCONFIG_LIBS CACHE STRING "Description")
+
+set(PRINTCHPLENV_COMMAND printchplenv --compiler --internal --simple)
+# message(STATUS "running ${PRINTCHPLENV_COMMAND} from ${CMAKE_CURRENT_SOURCE_DIR}/util")
+
+find_package(Python)
+
+execute_process(COMMAND ${CMAKE_CURRENT_SOURCE_DIR}/util/${PRINTCHPLENV_COMMAND}
```

Comment:
I think we need to set the non-internal of the above values, if they were set with `-D`, as environment variables for printchplenv.

Relatedly, I don't think we want to support `-D` of the internal variables or showing them in help. Putting them in the cmake cache sounds good though. Can we have cache variables that aren't in the help? Maybe we just put them in a separate section that says "don't set these".

Reply:
Oh, it looks like we can use `INTERNAL` for it. Go figure :)

Reply:
Regarding how to pass the environment variables, you would use `set( ENV )` -- see https://cmake.org/cmake/help/latest/command/set.html#set-environment-variable